src/components/Comments.js

```diff
@@ -0,0 +1,272 @@
+          this.setState({
+            error,
+          });
+        }
+      });
+    }
+  }
+
+  deleteComment(id: string) {
+    const jsonDoc = `{ "id" : "${id}", "mode" : "DELETE" }`;
+    this.context.searcher.search.addOrDeleteDocument(JSON.parse(jsonDoc), this.getComments);
+  }
+
+  populateComments() {
+    const { response } = this.state;
+    const loggedInUser = AuthUtils.getUserName(AuthUtils.getSavedUser());
+    if (response) {
+      const commentList = [];
+      response.documents.forEach((doc) => {
+        const comment = {};
+        comment.id = doc.getFirstValue('.id');
+        comment.text = doc.getFirstValue('comment_s');
+        comment.timestamp = doc.getFirstValue('timestamp_s');
+        comment.username = doc.getFirstValue('username_s');
+        const isCommentByLoggedInUser = (comment.username === loggedInUser);
+        const removeCommentOption = isCommentByLoggedInUser && (
+          <Glyphicon
+            glyph="trash"
+            onClick={() => {
+              this.deleteComment(comment.id);
+            }}
+            style={{ fontSize: '1em', color: '#003c7e', cursor: 'pointer' }}
+            title="Delete this comment"
+          />
+        );
+        commentList.push(
+          <div key={comment.id}>
+            <div
+              style={{ color: '#484848', backgroundColor: '#F5F5F5', borderRadius: '5px' }}
+              title={comment.timestamp}
+            >
+              <div style={{ padding: '0.5em' }}>
+                <b> <i> {comment.username} </i> </b>
+                <div style={{ float: 'right' }}>
+                  { removeCommentOption }
+                </div>
+                <br />
+                <span style={{ whiteSpace: 'pre-line' }}>
+                  {comment.text}
+                </span>
+              </div>
+            </div>
+            <br />
+          </div>,
+        );
+      });
+      this.setState({ commentList });
+    }
+  }
+
+  showCommentModal() {
+    this.setState({
+      showCommentModal: true,
+    });
+  }
+
+  hideCommentModal() {
+    this.setState({
+      showCommentModal: false,
+    });
+  }
+
+  captureComment(e: Event) {
+    if (e.target instanceof HTMLTextAreaElement) {
+      this.setState({
+        comment: e.target.value,
+      });
+    }
+  }
+
+  saveComment() {
+    const { docId } = this.props;
+    const { comment } = this.state;
+    const { searcher } = this.context;
+    const username = AuthUtils.getUserName(AuthUtils.getSavedUser());
+    const d = new Date();
+    const timestamp = `Posted on ${d.toLocaleDateString()} at ${d.toLocaleTimeString()}`;
+    const loggedDateTime = d.toISOString();
+    const id = username.concat(loggedDateTime);
+    // escape newline, carriage return and tab characters in the comment string
+    const formattedComment = comment.replace(/\n/g, '\\n').replace(/\r/g, '\\r').replace(/\t/g, '\\t');
+
+    const jsonDoc = `{ "fields" : { "comment_s" : [ "${formattedComment}" ], `
```

Comment:
You should be able to do this as an object - much easier to read, and use `JSON.stringify`.
```
const body = { 
  fields: {
    comment_s: [formattedComment],
    docId_s: [docId],
    username_s: [username],
    date: [loggedDateTime],
    timestamp_s: [timestamp],
    table: ["comments"]
  },
  id,
};
```